addons/mass_mailing/static/src/js/mass_mailing_wysiwyg.js

```diff
@@ -38,50 +38,30 @@ export class MassMailingWysiwyg extends Wysiwyg {
 
     /**
      * Sets SnippetsMenu fold state and switches toolbar.
-     * Instantiates a new floating Toolbar if needed.
+     * Configures the floating toolbar if needed.
      *
      * @param {Boolean} fold
      */
-    async setSnippetsMenuFolded(fold = true) {
-        if (fold) {
-            this.snippetsMenu.setFolded(true);
-            if (!this.floatingToolbar) {
-                // Instantiate and configure new toolbar.
-                const toolbarWrapper = new ComponentWrapper({}, Toolbar, this.state.toolbarProps);
-
-                // The wysiwyg can be instanciated inside an iframe. The dialog
-                // component is mounted on the global document.
-                const toolbarWrapperElement = document.createElement('div');
-                toolbarWrapperElement.style.display = 'contents';
-                document.body.append(toolbarWrapperElement);
-                await toolbarWrapper.mount(toolbarWrapperElement);
-                this.toolbarEl = toolbarWrapperElement.firstChild;
-                this.floatingToolbarEl = this.toolbarEl;
-
-                this._configureToolbar({ snippets: false });
-                this._updateEditorUI();
-                this.setCSSVariables(this.toolbarEl);
-                this.odooEditor.setupToolbar(this.toolbarEl);
-                if (this.odooEditor.isMobile) {
-                    document.body.querySelector('.o_mail_body').prepend(this.toolbarEl);
-                } else {
-                    document.body.append(this.toolbarEl);
-                }
+    setSnippetsMenuFolded(fold = true) {
+        this.snippetsMenu.setFolded(fold);
+        this.toolbarEl = fold ? this.floatingToolbarEl : this.snippetsMenuToolbarEl;
+        if (fold && !this.isFloatingToolbarReady) {
```

Comment:
1. If we rename `floatingToolbarEl` to `mainToolbarEl`, could we rename `isFloatingToolbarReady` to something like `_isMainToolbarReady`?
2. Could we also add a comment like
```
        // At the startup, the `SnippetMenu` setted its toolbar before the
        // `mainToolbarEl` had the chance to be configured so we configure it now
        // as we need it.
```

Reply:
I've pushed with the suggested changes.